File: Contest/Weekly Contest 309/Longest Nice Subarray/LongestNiceSubarray.kt
class Solution {
    fun longestNiceSubarray(nums: IntArray): Int {
        val n = nums.size

        var curr = 0
        var j = 0
        var ans = 0
        for (i in 0..n-1) {
            while (curr and nums[i] > 0) {
                curr = curr xor nums[j]
                j++
            }
            curr = curr or nums[i]
            ans = maxOf(ans, i-j+1)
        }

        return ans
    }
}

File: Problems/Algorithms/2401. Longest Nice Subarray/longest_nice_subarray.py
#!/usr/bin/env python3

from typing import List


class Solution:
    def longestNiceSubarray(self, nums: List[int]) -> int:
        used_bits = 0
        ans = 0

        start = 0
        for end in range(len(nums)):
            while used_bits & nums[end] != 0:
                used_bits ^= nums[start]
                start += 1

            used_bits |= nums[end]
            ans = max(ans, end - start + 1)

        return ans
